inst/include/interface/rcpp/rcpp_objects/rcpp_growth.hpp

```diff
@@ -53,117 +55,92 @@ std::map<uint32_t, GrowthInterfaceBase*> GrowthInterfaceBase::live_objects;
-
-    std::shared_ptr<fims::EWAAgrowth<TMB_FIMS_REAL_TYPE> > b0 =
-        std::make_shared<fims::EWAAgrowth<TMB_FIMS_REAL_TYPE> >();
-
-    // set relative info
-    b0->id = this->id;
-    b0->ewaa = make_map(this->ages, this->weights);  // this->ewaa;
-    // add to Information
-    d0->growth_models[b0->id] = b0;
-
-    // base model
-    std::shared_ptr<fims::Information<TMB_FIMS_FIRST_ORDER> > d1 =
-        fims::Information<TMB_FIMS_FIRST_ORDER>::GetInstance();
 
-    std::shared_ptr<fims::EWAAgrowth<TMB_FIMS_FIRST_ORDER> > b1 =
-        std::make_shared<fims::EWAAgrowth<TMB_FIMS_FIRST_ORDER> >();
+    virtual ~EWAAGrowthInterface() {
+    }
 
-    // set relative info
-    b1->id = this->id;
-    b1->ewaa = make_map(this->ages, this->weights);  // this->ewaa;
+    /** @brief get the id of the GrowthInterfaceBase object */
+    virtual uint32_t get_id() {
+        return this->id;
+    }
 
-    // add to Information
-    d1->growth_models[b0->id] = b1;
+    /**
+     * @brief Create a map of input numeric vectors
+     * @param weights T vector of weights
+     * @param ages T vector of ages
+     * @return std::map<T, T>
+     *
+     * */
+    inline std::map<double, double> make_map(std::vector<double> ages,
+            std::vector<double> weights) {
+        std::map<double, double> mymap;
+        for (uint32_t i = 0; i < ages.size(); i++) {
+            mymap.insert(std::pair<double, double>(ages[i], weights[i]));
+        }
+        return mymap;
+    }
 
-    // base model
-    std::shared_ptr<fims::Information<TMB_FIMS_SECOND_ORDER> > d2 =
-        fims::Information<TMB_FIMS_SECOND_ORDER>::GetInstance();
+    /** @brief Rcpp interface to the EWAAgrowth evaluate method
+     * you can call from R using
+     * ewaagrowth.evaluate(age)
+     * */
+    virtual double evaluate(double age) {
+        fims::EWAAgrowth<double> EWAAGrowth;
+
+        if (initialized == false) {
+            this->ewaa = make_map(this->ages, this->weights);
+            // Check that ages and weights vector are the same length
+            if (this->ages.size() != this->weights.size()) {
+                Rcpp::stop("ages and weights must be the same length");
+            }
+            initialized = true;
+        } else {
+            Rcpp::stop("this empirical weight at age object is already initialized");
+        }
+        EWAAGrowth.ewaa = this->ewaa;
+        return EWAAGrowth.evaluate(age);
+    }
+#ifdef TMB_MODEL
 
-    std::shared_ptr<fims::EWAAgrowth<TMB_FIMS_SECOND_ORDER> > b2 =
-        std::make_shared<fims::EWAAgrowth<TMB_FIMS_SECOND_ORDER> >();
+    template<typename T>
+    bool add_to_fims_tmb_internal() {
+        std::shared_ptr<fims::Information<T> > info =
+                fims::Information<T>::GetInstance();
 
-    // set relative info
-    b2->id = this->id;
-    b2->ewaa = make_map(this->ages, this->weights);  // this->ewaa;
+        std::shared_ptr<fims::EWAAgrowth<T> > data =
```

Comment:
the name `data` for the growth pointer is confusing. To clarify, rename to ewaagrowth. Another option is to rename this ewaa and rename the map (`ewaa` on line 125) to `ewaa_map`